server/src/main/java/io/crate/execution/ddl/tables/TransportCreateTableAction.java

```diff
@@ -92,49 +99,112 @@ protected CreateTableResponse read(StreamInput in) throws IOException {
 
     @Override
     protected ClusterBlockException checkBlock(CreateTableRequest request, ClusterState state) {
-        if (request.getCreateIndexRequest() != null) {
-            CreateIndexRequest createIndexRequest = request.getCreateIndexRequest();
-            return transportCreateIndexAction.checkBlock(createIndexRequest, state);
-        } else if (request.getPutIndexTemplateRequest() != null) {
-            PutIndexTemplateRequest putIndexTemplateRequest = request.getPutIndexTemplateRequest();
-            return transportPutIndexTemplateAction.checkBlock(putIndexTemplateRequest, state);
+        if (state.nodes().getMinNodeVersion().onOrAfter(Version.V_5_4_0)) {
+            if (request.partitionedBy().isEmpty()) {
+                return state.blocks().indexBlockedException(
+                    ClusterBlockLevel.METADATA_WRITE,
+                    request.getTableName().indexNameOrAlias()
+                );
+            } else {
+                return state.blocks().globalBlockedException(ClusterBlockLevel.METADATA_WRITE);
+            }
         } else {
-            throw new IllegalStateException("Unknown table request");
+            if (request.getCreateIndexRequest() != null) {
+                CreateIndexRequest createIndexRequest = request.getCreateIndexRequest();
+                return transportCreateIndexAction.checkBlock(createIndexRequest, state);
+            } else if (request.getPutIndexTemplateRequest() != null) {
+                PutIndexTemplateRequest putIndexTemplateRequest = request.getPutIndexTemplateRequest();
+                return transportPutIndexTemplateAction.checkBlock(putIndexTemplateRequest, state);
+            } else {
```

Comment:
Thanks! That's much simpler. 

I slightly modified isParitioned to `request.getPutIndexTemplateRequest() != null || request.partitionedBy().isEmpty() == false;` since I made `request.partitionedBy() ` not nullable (pass empty list in 5.3 code).